lib/Service/InitialStateService.php

```diff
@@ -110,5 +112,12 @@ private function provideOptions(): void {
 
 		$this->initialState->provideInitialState('theming-customLogo', $logo);
 		$this->initialState->provideInitialState('open_local_editor', $this->config->getAppValue(Application::APPNAME, 'open_local_editor', 'yes') === 'yes');
+		$fileName = $this->config->getUserValue($this->userId, Application::APPNAME, 'browsersetting');
+		if ($fileName) {
+			$browserSetting = $this->settingsService->getSettingsFile('userconfig/' . $this->userId,
```

Comment:
This is rather bad performance wise. File system access is rather slow and the initial state is built on every page load, so wondering if we shall rather write this to a memory cache

Reply:
You’re correct. We’re testing some features, and the patch will likely be discarded. If not, I’ll improve it. Thanks!